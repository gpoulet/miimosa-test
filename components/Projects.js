import { useQuery } from "@apollo/client";
import { ALL_PROJECTS_QUERY } from '../apollo'
import Project from "./Project"

function Projects() {
  const { loading, error, data } = useQuery(ALL_PROJECTS_QUERY);

  if (error)
    return <div>Error loading players.</div>;
  if (loading)
    return <div>Loading</div>;

  const { projects } = data;
  console.log("projects :", projects)

  return (
    <div>
      {projects.map(project => <Project {...project} key={project.id} />)}
    </div>
  )
}

export default Projects;