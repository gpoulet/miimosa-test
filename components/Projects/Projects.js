import Project from "../Project"

function Projects({ loading, error, data }) {

  if (error)
    return <div>Error loading projects.</div>;
  if (loading)
    return <div>Loading</div>;

  const { projects } = data;

  return (
    <div>
      {projects.map(project => <Project {...project} key={project.id} />)}
    </div>
  )
}

export default Projects;