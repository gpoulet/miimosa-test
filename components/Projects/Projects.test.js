import React from 'react';
import renderer from 'react-test-renderer';
import Projects from './Projects';

it('Projects error renders correctly', () => {
  const tree = renderer
    .create(<Projects error={true} loading={false} data={undefined} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Projects loading renders correctly', () => {
  const tree = renderer
    .create(<Projects error={false} loading={true} data={undefined}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


 const data = {
  "projects": [
    {
      "collected": 12,
      "description": "",
      "id": "2c714617-c83a-4675-99f4-e7e0d3cd6c92",
      "image_url": "",
      "status": "failed",
      "target": 1000,
      "title": "Boulangerie HiTekt",
      "user": {
        "first_name": "Daïsy",
        "last_name": "Artémis"
      }
    },
    {
      "collected": 15000,
      "description": "",
      "id": "c6d6f6f9-07d8-4540-87a7-3b8ff8869f55",
      "image_url": "",
      "status": "success",
      "target": 15000,
      "title": "Bar smoothie Bio",
      "user": {
        "first_name": "Lancelot",
        "last_name": "Little Finger"
      }
    },
    {
      "collected": 12000,
      "description": "",
      "id": "65c72f0b-8906-4504-a428-120b1a741905",
      "image_url": "",
      "status": "in_progress",
      "target": 15000,
      "title": "McDonald's sans conservateur",
      "user": {
        "first_name": "Hagrid",
        "last_name": "Potter"
      }
    },
  ]
}

it('Projects data renders correctly', () => {
  const tree = renderer
    .create(<Projects error={false} loading={false} data={data}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});