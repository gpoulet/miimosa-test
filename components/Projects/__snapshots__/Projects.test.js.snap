// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects data renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/2c714617-c83a-4675-99f4-e7e0d3cd6c92.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Boulangerie HiTekt
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        12€ collectés sur 1000€ (1%)
      </p>
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={1}
        className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-98.8%)",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a>
            Découvrir le projet
          </a>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/c6d6f6f9-07d8-4540-87a7-3b8ff8869f55.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Bar smoothie Bio
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        15000€ collectés sur 15000€ (100%)
      </p>
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={100}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(0%)",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a>
            Découvrir le projet
          </a>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root projectCard MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"/images/65c72f0b-8906-4504-a428-120b1a741905.png\\")",
          "height": 0,
          "paddingTop": "40%",
        }
      }
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        McDonald's sans conservateur
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        12000€ collectés sur 15000€ (80%)
      </p>
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={80}
        className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-20%)",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a>
            Découvrir le projet
          </a>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Projects error renders correctly 1`] = `
<div>
  Error loading projects.
</div>
`;

exports[`Projects loading renders correctly 1`] = `
<div>
  Loading
</div>
`;
